### PR TITLE
[DOC] Update Supported Algorithms page and add info about TSNE

### PR DESCRIPTION
**Pull request by outoftardis**
output:
- https://outoftardis.github.io/daal4py/algorithms.html
- https://outoftardis.github.io/daal4py/guide/acceleration.html

### CODE REVIEW

File: doc/sources/algorithms.rst [hunk 1]
```diff
@@ -20,177 +20,331 @@
...
+         all parameters except ``metric`` not in [`'euclidean'`, `'manhattan'`, `'minkowski'`, `'chebyshev'`, `'cosine'`]
+     - Sparse data is not supported
+   * - `train_test_split`
+     - All parameters are supported
+     - Only dense data is supported
+   * - `assert_all_finite`
+     - All parameters are supported
+     - Only dense data is supported
+   * - `pairwise_distance`
+     - With ``metric`` = `'cosine'` or `'correlation'`.
```
Comment (outoftardis) [inline doc/sources/algorithms.rst:197]: Can someone help me rephrase this in the same way as it's done for other algorithms?

File: doc/sources/algorithms.rst [hunk 2]
```diff
@@ -20,177 +20,331 @@
...
+     - Parameters
+     - Data formats
+   * - `RandomForestRegressor`
+     - All parameters are supported except:
+
+       - ``warm_start`` = `True`
+       - ``cpp_alpha`` != `0`
+       - ``criterion`` != `'mse'`
+       - ``oob_score`` = `True`
+       - ``sample_weight``
```
Comment (outoftardis) [inline doc/sources/algorithms.rst:271]: I've moved the info about sample_weight from the data column to the parameters column. question: should it say `sample_weight != None` or  it's ok as it is?

Comment (agorshk) [reply doc/sources/algorithms.rst:271]: Ok, let's align in all algorithms and use **sample_weight != None**, new place (in parameter column) is correct too

File: doc/sources/algorithms.rst (hunk 1 above)
Comment (agorshk) [reply doc/sources/algorithms.rst:197]: I suppose that @KalyanovD may help you

**Review by agorshk — APPROVED** [review]

File: doc/sources/algorithms.rst (hunk 2 above)
Comment (outoftardis) [reply doc/sources/algorithms.rst:271]: > Ok, let's align in all algorithms and use **sample_weight != None**, new place (in parameter column) is correct too

ok, I'll make these updates

File: doc/sources/algorithms.rst [hunk 3]
```diff
@@ -20,177 +20,331 @@
...
+
+.. list-table::
+   :widths: 10 30 20
+   :header-rows: 1
+   :align: left
+
+   * - Algorithm
+     - Parameters
+     - Data formats
+   * - `NearestNeighbors`
```
Comment (KalyanovD) [inline doc/sources/algorithms.rst:181]: "Other tasks" -> "Unsupervised learning" for NearestNeighbors?
https://scikit-learn.org/stable/modules/neighbors.html#unsupervised-nearest-neighbors

File: doc/sources/algorithms.rst (hunk 1 above)
Comment (KalyanovD) [reply doc/sources/algorithms.rst:197]: "All parameters except metric != cosine and correlation"?

File: doc/sources/algorithms.rst (hunk 3 above)
Comment (outoftardis) [reply doc/sources/algorithms.rst:181]: clustering and dimensionality reduction would also be considered unsupervised learning, and so it doesn't make sense to have unsupervised learning as a separate category of tasks if we already have clustering and dimensionality reduction categories.

should there be a separate category for this algorithm? "Nearest Neighbors", the same wat we have in oneDAL docs:

image

File: doc/sources/algorithms.rst (hunk 1 above)
Comment (outoftardis) [reply doc/sources/algorithms.rst]: I've decided to use "not in" phrase

File: doc/sources/algorithms.rst (hunk 2 above)
Comment (outoftardis) [reply doc/sources/algorithms.rst]: done

File: doc/sources/algorithms.rst [hunk 4]
```diff
@@ -20,177 +20,333 @@
...
+     
+       - ``metric`` not in [`'cosine'`, `'correlation'`]
+     - Only dense data is supported
+   * - `roc_auc_score`
+     - All parameters are supported except:
+       
+       - ``average``
+       - ``sample_weight`` != `None`
+       - ``max_fpr``
+       - ``multi_class``
```
Comment (outoftardis) [inline doc/sources/algorithms.rst:207]: @agorshk similar question to the previous one about sample_weights: can I use `<parameter name> != None` for other parameters here? or should there be other values specified?

Comment (agorshk) [reply doc/sources/algorithms.rst:207]: yes, let's use <parameter name> != None

Comment (outoftardis) [reply doc/sources/algorithms.rst:207]:
```suggestion
       - ``average`` != `None`
       - ``sample_weight`` != `None`
       - ``max_fpr`` != `None`
       - ``multi_class`` != `None`
```

File: doc/sources/algorithms.rst (hunk 3 above)
Comment (outoftardis) [reply doc/sources/algorithms.rst:181]: @KalyanovD  do you agree with the suggestion?

**Review by Pahandrovich — APPROVED** [review]
